[Serializer] improve documentation of default_context for the serializer

dbu opened the pull request:
additional info for https://github.com/symfony/symfony-docs/pull/17310

fix #17306

the code example will only work once https://github.com/symfony/symfony/pull/47243 is merged

dbu committed:
document default context for the serializer

dbu commented:
build should get green when retrying, as the code is now merged to 6.2.

OskarStark commented on reference/configuration/framework.rst:
```diff
@@ -2743,6 +2743,18 @@ paths
+to a `json_encode flags bitmask`_.
+
+You can inspect the :ref:`serializer context builders <serializer-using-context-builders>`
+to discover the available settings.
```
```suggestion
to discover the available settings.

.. versionadded:: 6.2

    The ``default_context`` parameter was introduced in Symfony 6.2.
```

dbu replied:
that is not correct. default_context has been added in symfony 5.4 but was not documented aparently. 
https://github.com/symfony/symfony/blob/5.4/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L1010

dbu replied:
the context builders seem to have been added in 6.1 https://github.com/symfony/symfony/tree/6.1/src/Symfony/Component/Serializer/Context

OskarStark replied:
Oh ok, in this case I would vote for adding the default context parameter against `5.4` branch, and the yaml stuff against `6.2`

dbu replied:
right. i made https://github.com/symfony/symfony-docs/pull/17310

i will remove the changes from this MR, as they don't clash

dbu replied:
actually the part about context builders was only added later, in 6.1. i will rebase this on 6.2 when you merged upstream, to keep the reference.

best would be if all the context settings would be documented, but i don't have time to do that.

OskarStark replied:
I merged and unmerged your other PR, thanks!

dbu replied:
thanks. rebased and solved the conflicts.

dbu commented on reference/configuration/framework.rst:
```diff
@@ -2743,6 +2743,22 @@ paths
+You can inspect the :ref:`serializer context builders <serializer-using-context-builders>`
+to discover the available settings.
+
+.. versionadded:: 6.2
```
```suggestion
.. versionadded:: 5.4
```

dbu commented on reference/configuration/framework.rst:
```diff
@@ -2743,6 +2743,22 @@ paths
+
+.. versionadded:: 6.2
+
+    The ``default_context`` parameter was introduced in Symfony 6.2.
```
```suggestion
    The ``default_context`` parameter was introduced in Symfony 5.4.
```

dbu commented on serializer.rst:
```diff
@@ -130,14 +130,15 @@ configuration:
             serializer:
                 default_context:
                     enable_max_depth: true
+                    yaml_indentation: 2
```
yaml_indentation however has been added in 6.2 only

OskarStark commented:
Please also adjust the title of the PR after rebasing, thanks

dbu commented:
rebased, cleaned up and adjusted the title.

regarding the linter failure: looks like thats broken in all 6.x versions: https://github.com/symfony/symfony-docs/actions/runs/3160768594/jobs/5145596113

can you fix that in 6.0 and then merge it up to 6.1 and 6.2? or we just merge with the known error...

dbu commented on serializer.rst:
```diff
@@ -146,11 +147,13 @@ configuration:
                 ->defaultContext([
-                    AbstractObjectNormalizer::ENABLE_MAX_DEPTH => true
+                    AbstractObjectNormalizer::ENABLE_MAX_DEPTH => true,
+                    YamlEncoder::YAML_INDENTATION => 2,
```
there is a build failure about this, but i don't understand it.

OskarStark replied:
any idea @Nyholm

OskarStark commented on reference/configuration/framework.rst:
```diff
@@ -2752,6 +2752,9 @@ A map with default context options that will be used with each ``serialize`` and
+to discover the available settings.
+
 .. versionadded:: 5.4
 
```
Ro remove those messages, please rebase again, thanks

dbu replied:
thanks, rebased.
